climada/hazard/test/test_trop_cyclone.py

```diff
@@ -503,29 +504,34 @@ def test_apply_criterion_track(self):
         intensity[3, 3] = 3
         tc = TropCyclone(
             intensity=sparse.csr_matrix(intensity),
-            basin=['NA', 'NA', 'NA', 'NO'],
+            basin=['NA', 'NA', 'NA', 'WP'],
             category=np.array([2, 0, 4, 1]),
             event_id=np.arange(4),
             frequency=np.ones(4) * 0.5,
+            date=np.array([723795, 728395, 738395, 724395])
         )
 
-        tc_cc = tc.apply_climate_scenario_knu(ref_year=2050, rcp_scenario=45)
-        self.assertTrue(np.allclose(tc.intensity[1, :].toarray(), tc_cc.intensity[1, :].toarray()))
-        self.assertTrue(np.allclose(tc.intensity[3, :].toarray(), tc_cc.intensity[3, :].toarray()))
+        tc_cc = tc.apply_climate_scenario_knu(percentile='50',
+                                              scenario='4.5',
+                                              target_year=2050)
+        self.assertFalse(
+            np.allclose(tc.frequency[1], tc_cc.frequency[1])
+            )
         self.assertFalse(
-            np.allclose(tc.intensity[0, :].toarray(), tc_cc.intensity[0, :].toarray()))
+            np.allclose(tc.frequency[0], tc_cc.frequency[0])
+        )
         self.assertFalse(
-            np.allclose(tc.intensity[2, :].toarray(), tc_cc.intensity[2, :].toarray()))
-        self.assertTrue(np.allclose(tc.frequency, tc_cc.frequency))
+            np.allclose(tc.frequency[2], tc_cc.frequency[2])
+            )
+        self.assertFalse(
+            np.allclose(tc.frequency[3], tc_cc.frequency[3])
+            )
```

Comment:
These tests check that the frequencies are ... not the same? Why are intensities not checked anymore? What does the "not the same" check achieve? These tests should check the actual results. 

Please always use `np.testing.assert_allclose(x, y)` instead of `self.assertTrue(np.allclose(x, y))` to get a useful error message when the test fails.

Reply:
Yes I am testing that the module actually have changed the frequencies, which is essence all it does. Intensities are _directly_ changed anymore, because that would double counting. The test after this one actually tests that the operations that change these frequencies work.

Could you please show me how to use `np.testing.assert_allclose(x, y)` in the negative case? simply with a tilde?

Reply:
Your tests should not check that the values have changed, but should instead check the resulting values. For that reason, there is no negative case of `np.testing.assert_allclose`. Note that such a negative case would even succeed of one of the arrays suddenly turned into NaNs.